pkg/virt-handler/vm.go

```diff
@@ -3103,71 +3103,100 @@ func (c *VirtualMachineController) handleStartingVMI(
 		return false, fmt.Errorf(failedDetectIsolationFmt, err)
 	}
 
+	if err := c.setupDevicesOwnerships(vmi, isolationRes); err != nil {
+		return false, err
+	}
+
+	return true, nil
+}
+
+func (c *VirtualMachineController) setupDevicesOwnerships(vmi *v1.VirtualMachineInstance, isolationRes isolation.IsolationResult) error {
 	virtLauncherRootMount, err := isolationRes.MountRoot()
 	if err != nil {
-		return false, err
+		return err
 	}
 
 	err = c.claimDeviceOwnership(virtLauncherRootMount, "kvm")
 	if err != nil {
-		return false, fmt.Errorf("failed to set up file ownership for /dev/kvm: %v", err)
+		return fmt.Errorf("failed to set up file ownership for /dev/kvm: %v", err)
 	}
+
 	if virtutil.IsAutoAttachVSOCK(vmi) {
 		if err := c.claimDeviceOwnership(virtLauncherRootMount, "vhost-vsock"); err != nil {
-			return false, fmt.Errorf("failed to set up file ownership for /dev/vhost-vsock: %v", err)
+			return fmt.Errorf("failed to set up file ownership for /dev/vhost-vsock: %v", err)
+		}
+	}
+
+	if err := c.configureHostDisks(vmi, isolationRes, virtLauncherRootMount); err != nil {
+		return err
+	}
+
+	if err := c.configureSEVDeviceOwnership(vmi, isolationRes, virtLauncherRootMount); err != nil {
+		return err
+	}
+
+	if virtutil.IsNonRootVMI(vmi) {
+		if err := c.nonRootSetup(vmi); err != nil {
+			return err
+		}
+	}
+
+	if err := c.configureVirtioFS(vmi, isolationRes); err != nil {
+		return err
+	}
+
+	// set runtime limits as needed
+	err = c.podIsolationDetector.AdjustResources(vmi, c.clusterConfig.GetConfig().AdditionalGuestMemoryOverheadRatio)
+	if err != nil {
+		return fmt.Errorf("failed to adjust resources: %v", err)
+	}
+
+	if util.IsSEVAttestationRequested(vmi) {
+		sev := vmi.Spec.Domain.LaunchSecurity.SEV
+		if sev.Session == "" || sev.DHCert == "" {
+			// Wait for the session parameters to be provided
+			return nil
```

Comment:
How does this wait if this returns nil? 
@vladikr

Reply:
Because since nothing is provided we will go to another sync loop. See where this is called.

Reply:
Well we return nil all the time as far I see so we always return, ready to proceed. Before we would actually wait since we returned false as not ready in case the session and/or dhcert was empty. 

unfortunately I don't see unit test which would catch this

Reply:
Hmm, yes, you're right. that's a problem!